### PR TITLE
track dom-slot created MutationObservers to afford teardown of an Arc

### PR DESCRIPTION
**Pull request by sjmiles**
We'll probably want something more sophisticated when Maria gets back, but I wanted to get this in so we could keep testing serialization loop-back.

If y'all would rather wait for a more comprehensive change, that's ok too.

### CODE REVIEW

File: runtime/arc.js [hunk 1]
```diff
@@ -209,7 +209,7 @@ ${this.activeRecipe.toString()}`;
       pecFactory,
       loader,
       storageProviderFactory: manifest._storageProviderFactory,
-      context: manifest
+      context
```
Comment (shaper) [inline runtime/arc.js]: Is this intentional? It looks not related to DOM slots.

Comment (sjmiles) [reply runtime/arc.js]: It's necessary for proper round-trip testing. Fwiw, I checked with @shans before I pushed it.
